brooklyn-server/utils/common/src/test/java/org/apache/brooklyn/util/text/IdentifiersTest.java

```diff
@@ -98,5 +96,17 @@ public void testJavaClassRegex() {
         Assert.assertTrue("foo".matches(Identifiers.JAVA_GOOD_BINARY_REGEX));
         Assert.assertTrue("foo.bar.Baz$1".matches(Identifiers.JAVA_GOOD_BINARY_REGEX));
     }
+
+    @Test
+    public void testPassword() {
+        String upper = "ABC";
+        String numbers = "123";
+        String symbols = "!£$";
+        String id1 = Identifiers.makeRandomPassword(4, upper, numbers, symbols, Identifiers.PASSWORD_VALID_CHARS);
```

Comment:
anyone inspecting the code will be in a position to quite easily reconstruct passwords.
the `makeRandomPassword` method could simply randomly reorder the result, and the test here check for containment rather than position.